packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -513,6 +517,32 @@ class Treemap extends StatelessWidget {
       },
     );
   }
+
+  void _onTapDown(
+    TapDownDetails details,
+    List<PositionedCell> positionedCells,
+  ) {
+    final x = details.localPosition.dx;
+    final y = details.localPosition.dy;
+    // Define target point as rectangle with width and height as 0.01 to
+    // represent the touch point.
+    final touchPoint = Rect.fromLTWH(x, y, 0.01, 0.01);
+    for (final cell in positionedCells) {
+      final cellRect = Rect.fromLTWH(
+        cell.left,
+        cell.top,
+        cell.width,
+        cell.height,
+      );
+      // TODO(peterdjlee): Optimize with better algorithm to find the overlapping cell.
+      if (touchPoint.overlaps(cellRect)) {
+        // Ensure the new root has children to show when zoomed into.
+        if (cell.node.children.isNotEmpty) {
```

Comment:
This special case seems a little surprising. Why not let users zoom all the way in?
If we don't want to we should select the parent of the selected node instead.

Reply:
For now I made it select the parent of the node, but will take a look at just zooming down to a single cell. Might need to shift around the code a bit.